internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -354,6 +354,52 @@ type networkLoadBalancerConfig struct {
 	appDNSName           *string
 }
 
+func getPortValues(cfg manifest.SecurityGroupRule) (fromPort int, toPort int, err error) {
+	if cfg.PortsConfig.Ports.IsEmpty() {
+		toPort = *cfg.PortsConfig.Port
+		fromPort = *cfg.PortsConfig.Port
+	} else {
+		fromPort, toPort, err = cfg.PortsConfig.Ports.Parse()
+		if err != nil {
+			return 0, 0, err
+		}
+	}
+	return fromPort, toPort, nil
+}
```

Comment:
What do you think of moving this to be a method on `*manifest.SecurityGroupRule` instead? This way `IsEmpty` and `Parse` can become private methods.
```go
// Ports returns the from and to ports of a security group rule.
func (r manifest.SecurityGroupRule) Ports() (from, to int, err error) {
  if r.PortsConfig.Ports.isEmpty() {
    return *cfg.PortsConfig.Port, *cfg.PortsConfig.Port, nil // a single value is provided for ports. 
  }
  return cfg.PortsConfig.Ports.parse()
}
```

Reply:
That's such a great idea. I like this one. I will make the change. :)